pkg/api/handlers/setup/handler.go

```diff
@@ -0,0 +1,54 @@
+package setup
+
+import (
+	"net/http"
+
+	"github.com/obot-platform/obot/apiclient/types"
+	"github.com/obot-platform/obot/pkg/api"
+	"github.com/obot-platform/obot/pkg/gateway/client"
+	gwtypes "github.com/obot-platform/obot/pkg/gateway/types"
+)
+
+type Handler struct {
+	gatewayClient *client.Client
+}
+
+func NewHandler(gc *client.Client) *Handler {
+	return &Handler{
+		gatewayClient: gc,
+	}
+}
+
+// requireBootstrap checks if the request is from the bootstrap user.
+// Returns an error if not authenticated as bootstrap.
+func (h *Handler) requireBootstrap(req api.Context) error {
+	// Check if user is bootstrap user
+	if req.User.GetName() != "bootstrap" {
+		return types.NewErrHTTP(http.StatusForbidden,
+			"this endpoint requires bootstrap authentication")
+	}
+	return nil
+}
+
+// requireBootstrapEnabled checks if bootstrap mode is enabled.
+// Returns 404 if bootstrap is disabled.
+func (h *Handler) requireBootstrapEnabled(req api.Context) error {
+	// Query all Owner users
+	adminUsers, err := h.gatewayClient.Users(req.Context(), gwtypes.UserQuery{
+		Role: types.RoleOwner,
+	})
+	if err != nil {
+		return err
+	}
+
+	// Check if any non-bootstrap Owner with email exists
+	for _, u := range adminUsers {
+		if u.Username != "bootstrap" && u.Email != "" {
```

Comment:
Do we also want to pay attention to the "force bootstrap" environment variable?

Reply:
So I thought about this, and my hunch is no. This is more for first-time setup, while the force bootstrap mode is more for recovery purposes if something was configured wrong. Let me know if you disagree.